ui/components/multichain/asset-picker-amount/asset-picker-amount.tsx

```diff
@@ -1,105 +1,51 @@
-import { PRIMARY } from '../../../helpers/constants/common';
-import TokenBalance from '../../ui/token-balance';
-import { getSelectedIdentity } from '../../../selectors';
 import MaxClearButton from './max-clear-button';
-import AssetPicker from './asset-picker/asset-picker';
-
-const renderCurrencyInput = (asset: Asset, amount: Amount) => {
-  const dispatch = useDispatch();
-  const t = useI18nContext();
+import {
+  AssetPicker,
+  type AssetPickerProps,
+} from './asset-picker/asset-picker';
+import { SwappableCurrencyInput } from './swappable-currency-input/swappable-currency-input';
+import { AssetBalance } from './asset-balance/asset-balance';
+import useIsFiatPrimary from './hooks/useIsFiatPrimary';
 
-  if (asset.type === AssetType.native) {
-    return (
-      <>
-        <UserPreferencedCurrencyInput
-          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-          // @ts-ignore: I'm not sure why the types don't find `onChange`
-          onChange={(newAmount: string) =>
-            dispatch(updateSendAmount(newAmount))
-          }
-          hexValue={amount.value}
-          className="asset-picker-amount__input"
-          swapIcon={(onClick: React.MouseEventHandler) => (
-            <ButtonIcon
-              backgroundColor={BackgroundColor.transparent}
-              iconName={IconName.SwapVertical}
-              ariaLabel={t('switchInputCurrency')}
-              size={ButtonIconSize.Sm}
-              color={IconColor.primaryDefault}
-              onClick={onClick}
-            />
-          )}
-        />
-      </>
-    );
-  }
-  if (asset.type === AssetType.NFT) {
-    return (
-      <>
-        <Box marginLeft={'auto'}>
-          <Text variant={TextVariant.bodySm}>{t('tokenId')}</Text>
-          <Text
-            variant={TextVariant.bodySm}
-            fontWeight={FontWeight.Bold}
-            marginLeft={10}
-          >
-            {asset?.details?.tokenId}
-          </Text>
-        </Box>
-      </>
-    );
-  }
-
-  return (
-    <UserPreferencedTokenInput
-      onChange={(newAmount: string) => dispatch(updateSendAmount(newAmount))}
-      token={asset.details}
-      value={amount.value}
-      className="asset-picker-amount__input"
-    />
-  );
-};
+interface AssetPickerAmountProps extends AssetPickerProps {
+  // all of these props should be explicitly received
+  asset: Asset;
+  amount: Amount;
+  onAmountChange: (newAmount: string) => void;
+}
 
 // A component that combines an asset picker with an input for the amount to send.
-export const AssetPickerAmount = () => {
+export const AssetPickerAmount = ({
+  asset,
+  amount,
+  onAmountChange,
+  ...assetPickerProps
+}: AssetPickerAmountProps) => {
   const t = useI18nContext();
-  const { asset, amount } = useSelector(getCurrentDraftTransaction);
+
   const selectedAccount = useSelector(getSelectedIdentity);
 
+  const isFiatPrimary = useIsFiatPrimary();
+
+  const [isFocused, setIsFocused] = useState(false);
```

Comment:
Can the `:focus` scss pseudo-class be used to set borderColor instead?

Reply:
It's `focus-within` since it's a wrapper (🤓☝️), but I don't think we can use it anyways since the error state in that component needs to take priority.